core/tests/puppeteer-acceptance-tests/user-utilities/blog-post-admin-utils.js

```diff
@@ -212,6 +236,28 @@ module.exports = class e2eBlogPostAdmin extends baseUser {
     }, blogPostTitle);
   }
 
+  /**
+   * This function checks that the publish button is disabled.
+   * @param {string} expectedWarningMessage - The warning message to be
+   * displayed when the user tries to publish a duplicate blog post.
+   */
+  async expectUserUnableToPublishBlogPost(expectedWarningMessage) {
+    const toastMessageBox = await this.page.$(
+      'div.e2e-test-toast-warning-message');
+    const toastMessageWarning = await this.page.evaluate(
+      el => el.textContent, toastMessageBox);
+    const isPublishButtonDisabled = await this.page.$eval(
+      publishBlogPostButton, (button) => {
+        return button.disabled;
+      });
+    if (!isPublishButtonDisabled ||
+        expectedWarningMessage !== toastMessageWarning) {
+      throw new Error('User is able to publish duplicate blog post');
+    }
+
+    showMessage('User is not able to publish blog post if title duplicates');
```

Comment:
@Shivkant-Chauhan See my previous comment. This function should be generic and should not be special-cased/hardcoded to a particular reason for not being able to publish. The passed-in argument should be used to determine that, instead.

Reply:
can you please check if now the function is more generic about checking that user is unable to publish blog post.